portfolio/src/main/webapp/script.js

```diff
@@ -28,10 +28,25 @@ function addRandomLanguage() {
 }
 
 /**
- * Adds a greeting with my name to the page.
+ * Appends messages in servlet as children list elements to the page.
  */
-async function showContent() {
-  const response = await fetch('/data');
-  const content = await response.text();
-  document.getElementById('content-container').innerText = content;
+function showContent() {
+  fetch('/data').then(response => response.json()).then((content) => {
+    const messagesElement = document.getElementById('content-container');
+    messagesElement.innerHTML = '';
+    const contentLength = content.length;
+    let i;
+    for(i = 1; i < contentLength+1; i++){
```

Comment:
There's a few ways you can consider consolidating this:

```suggestion
// Option 1: Using the javascript for-of statement (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of), but this will not include the counter i. You could probably include a counter though if you wanted to use this option.
    for (const message of content) {
      messagesElement.appendChild(createListElement('Quote:  ' + message));
    }
    
// Option 2: Keep using a for loop, but have it 0 indexed. Javascript arrays are 0 indexed, so you'll typically see 0 index values are preferred, especially because it tends to be less error-prone.
    for (let i = 0; i < content.length; ++i ) {
      messagesElement.appendChild(createListElement('Quote #' + (i + 1) + ':  ' + content[i]));
    }
```

Reply:
Done!